app/src/main/java/fr/neamar/kiss/preference/AddSearchProviderPreference.java

```diff
@@ -154,10 +153,10 @@ private boolean validate() {
             //cancel close dialog
             return false;
         }
-        //check if a valid url is given
-        if (!validateUrl()) {
-            Toast.makeText(this.getContext(), R.string.search_provider_error_url, Toast.LENGTH_SHORT).show();
-            //not a url
+        //check if a valid uri is given
+        if (!validateUri()) {
```

Comment:
IMO `validateUrl` should still be checked first and only if it is no valid url, `validateUri` should be called.
`validateUri` will capture url too, but behaves a little different as it will also check if there is an app that can handle uri.